### ISSUE TITLE
Not all dependencies required to build jsweet are available on Maven Central

### ISSUE THREAD

**Issue by kohlschuetter** [cincheo/jsweet#764]
Unfortunately, at this point not all required components are available on Maven Central.

Currently, components required to build jsweet, such as `org.cincheo:file-visitor:1.0.0` and `org.jsweet.ext:typescript.java-ts.core:2.0.4`, as well as all `org.jsweet.candies:*.*` are only available at http://repository.jsweet.org.

That's a roadblock for any continued development of jsweet by third parties, as they are at the mercy of the availability of that custom repository.

jsweet should at least publish all artifacts required to build jsweet (which include some candies for unit testing) to Maven Central.

Related to https://github.com/cincheo/jsweet/issues/565

**Comment by winrid** [comment]
This is a bit higher priority now that the jsweet.org repo is down :)

**Comment by renaudpawlak** [comment]
Indeed, now that the repo is down, it becomes high prio. It requires further investigation because I am not sure that file-visitor and typescript.java-ts.core are actually required... I don't have time to work on this right now so any help would be appreciated.

**Comment by hoshie82** [comment]
snippet of the jsweet-transpiler pom file
````
		<dependency>
			<groupId>org.jsweet.ext</groupId>
			<artifactId>typescript.java-ts.core</artifactId>
			<version>2.0.3</version>
		</dependency>
		<dependency>
			<groupId>org.jsweet.ext</groupId>
			<artifactId>sourcemap-builder</artifactId>
			<version>1.0.0</version>
		</dependency>
````
was only available at jsweet repository.

**Comment by kevindeyne-ab** [comment]
Having all these on maven central would definitely help.

For what it's worth, this repository does contain the extension
https://github.com/cincheo/jsweet/tree/develop/typescript.java-ts.core
which, depending on your usecase, could be enough to just build that locally/push onto your own artifact store. 

Same thing with the sourcemap-builder, though I'm not sure if there's a repo of this somewhere. I do still have the downloaded jar, in case that can help someone forward. I'll share it here:

[sourcemap-builder-1.0.0.zip](https://github.com/user-attachments/files/16971286/sourcemap-builder-1.0.0.zip)

**Comment by vorth** [comment]
I'm going to take a look at removing the dependencies, if Renaud is correct, since I build from a fork regularly anyway.  Anyone else want to collaborate?

Hmmm... it appears that @kohlschuetter has already done significant work on dependencies, so I'm not clear on why the loss of Artifactory is even a problem.  Christian, do you mind summarizing your work from last year for me?  There are a lot of commits to read.  I suspect I may be using the wrong command-line to build.

**Comment by vorth** [comment]
The `develop` branch is more self-contained, thanks to work by @kohlschuetter last year.  However, there is still a dependency on `sourcemap-builder` in Artifactory.  It looks like it may be possible to pre-build and install using [this repo](https://github.com/lgrignon/sourcemap-builder).

**Comment by lgrignon** [comment]
Hello all,

FYI typescript.java-ts.core is required in order to make https://github.com/cincheo/jsweet/blob/develop/transpiler/src/main/java/org/jsweet/transpiler/TypeScript2JavaScriptWithTsserverTranspiler.java work

Would it be helpful for you to take over if we keep the Artifactory up for one more week?

**Comment by j75** [comment]
I manually installed **sourcemap-builder-1.0.0.jar**, now trying to rebuild the whole projects fails with
```
[ERROR] Failed to execute goal on project jsweet-transpiler: Could not resolve dependencies for project 
org.jsweet:jsweet-transpiler:jar:4.0.0-SNAPSHOT: The following artifacts could not be resolved: 
org.jsweet.candies:jquery:jar:1.10.0-20170726, org.jsweet.candies:jqueryui:jar:1.11.0-20170726, 
org.jsweet.candies:backbone:jar:1.3.0-20170726, org.jsweet.candies:angular:jar:1.4.0-20170726, 
org.jsweet.candies:angular-route:jar:1.2.0-20170726, org.jsweet.candies:node:jar:7.5.0-20170726, 
org.jsweet.candies:express:jar:4.14.0-20170726, org.jsweet.candies:express-errorhandler:jar:4.14.0-20170726, org.jsweet.candies:express-body-parser:jar:4.14.0-20170726, 
org.jsweet.candies:socket.io:jar:1.4.5-20170726, org.jsweet.candies:threejs:jar:75-20171101: Could not find artifact org.jsweet.candies:jquery:jar:1.10.0-20170726 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```
Where could all these jars be found ?

**Comment by FlySkyPie** [comment]
I just discover there is mirror have candies:
https://mavenrepo.ktbyte.com/org/jsweet/candies/


about `jsweet-typescript.java-ts.core:2.0.4`, I trying made a fork:
https://github.com/FlySkyPie/jsweet-typescript.java-ts.core

should able use it by download from jitpack like this:
```xml
	<repositories>
		<repository>
		    <id>jitpack.io</id>
		    <url>https://jitpack.io</url>
		</repository>
	</repositories>
	<dependencies>		
		<dependency>
			<groupId>com.github.FlySkyPie</groupId>
			<artifactId>jsweet-typescript.java-ts.core</artifactId>
			<version>2.0.4</version>
		</dependency>
	</dependencies>
```

**Comment by kohlschuetter** [comment]
@FlySkyPie This is just another repository that is not Maven-central, which does not resolve the issue. It just moves the point-of-failure somewhere else.

Also note that configuring your POM with a broad, not project-specific repository like jitpack further increases the chance of a security vulnerability by sourcing unrelated material from an untrusted / malicious source.
